src/svgdom/util/element_by_id_finder.cpp

```diff
@@ -0,0 +1,114 @@
+#include "element_by_id_finder.hpp"
+
+#include <utki/debug.hpp>
+
+#include "../visitor.hpp"
+
+using namespace svgdom;
+
+namespace{
+class cache_creator : virtual public svgdom::const_visitor{
+public:
+	std::unordered_map<std::string, const element*> cache;
+
+	void add_to_cache(const svgdom::element& e){
+		if(!e.id.empty()){
+			this->cache.insert(std::make_pair(e.id, &e));
+		}
+	}
+	
+	void visit_container(const svgdom::element& e, const svgdom::container& c, const svgdom::styleable& s){
```

Comment:
in sthis case we are not interested in `styleable` part of the element, thus, the `styleable` argument can be removed from this function. And then, at a first glance it looks to me that the visitor becomes same as standard visitor. I mean that in this case you don't have to override every `visit()` method of the class, maybe only overriding of `default_visit()` could be enough, thus shortening the code significantly. Please have a look if that is possible in this case.